core/src/main/java/io/github/belgif/rest/guide/validator/output/JsonOutputProcessor.java

```diff
@@ -69,9 +72,20 @@ private void writeToFile(ViolationReport violationReport) {
         mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
         mapper.enable(SerializationFeature.INDENT_OUTPUT);
         try {
-            mapper.writeValue(new File(output, FILE_NAME), violationReport);
+            mapper.writeValue(resolveOutputFile(), violationReport);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
+
+    public File resolveOutputFile() {
+        if (outputFile == null) {
+            return new File(outputDirectory, DEFAULT_FILE_NAME).getAbsoluteFile();
+        }
+        if (!outputFile.getName().endsWith(".json")) {
+            outputFile = new File(outputFile.getPath() + ".json");
```

Comment:
I wouldn't automagically add a file extension, such things are only for GUI save dialogs.